### ISSUE TITLE
run_pt.sh 4卡训练，训练数据打印问题

### ISSUE THREAD

**Issue by cqray1990** [shibing624/MedicalGPT#469]
### Describe the bug
Please provide a clear and concise description of what the bug is. If applicable, add screenshots to help explain your problem, especially for visualization related problems.

Running tokenizer on dataset (num_proc=10): 100%|█| 369895/369895 [11:06<00:00, 
Running tokenizer on dataset (num_proc=10): 100%|█| 500/500 [00:19<00:00, 25.60 
4卡训练，per_device_train_batch_size = 16，gradient_accumulation_steps=8 训练数据总共369895，测试数据500，可是打印日子只有：

 __main__:main:585 - Num train_samples: 141331

2025-08-13 16:03:09.592 | DEBUG    | __main__:main:599 - Num eval_samples: 175
4张卡打印了4次，为什么训练数据日志是 141331条数据，乘以4也不是369895，测试数据175乘以4也不是500

**Comment by shibing624** [comment]
有max_prompt_len, max_answer_len的要求，不符合的会过滤，检查下参数设置

**Comment by cqray1990** [comment]
> max_answer_len
这些参数都没设置，单卡跑的时候能打印对，怎么都算不对
、main:main:585 - Num train_samples: 36985

训练参数：

CUDA_VISIBLE_DEVICES=4,5,6,7 torchrun --nproc_per_node 4 pretraining.py \
    --model_name_or_path Qwen2.5-0.5B-Instruct \
    --train_file_dir medical/pretrain/train \
    --validation_file_dir medical/pretrain/valid \
    --per_device_train_batch_size 16 \
    --per_device_eval_batch_size 4 \
    --do_train \
    --do_eval \
    --use_peft True \
    --seed 42 \
    --max_train_samples -1 \
    --max_eval_samples -1 \
    --num_train_epochs 1 \
    --learning_rate 2e-4 \
    --warmup_ratio 0.05 \
    --weight_decay 0.01 \
    --logging_strategy steps \
    --logging_steps 10 \
    --eval_steps 50 \
    --eval_strategy steps \
    --save_steps 1000 \
    --save_strategy steps \
    --save_total_limit 13 \
    --gradient_accumulation_steps 8 \
    --preprocessing_num_workers 10 \
    --block_size 1024 \
    --group_by_length True \
    --output_dir outputs-pt-qwen25-v1 \
    --overwrite_output_dir \
    --ddp_timeout 30000 \
    --logging_first_step True \
    --target_modules all \
    --lora_rank 8 \
    --lora_alpha 16 \
    --lora_dropout 0.05 \
    --torch_dtype bfloat16 \
    --bf16 \
    --device_map auto \
    --report_to "swanlab" \
    --ddp_find_unused_parameters False \
    --gradient_checkpointing True \
    --cache_dir ./cache
